pkg/reconciler/v1alpha1/autoscaling/autoscaling.go

```diff
@@ -170,6 +175,48 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 
 func (c *Reconciler) reconcile(ctx context.Context, key string, kpa *kpa.PodAutoscaler) error {
 	logger := logging.FromContext(ctx)
+	logger.Debugf("PodAutoscaler class %q", kpa.Class())
+
+	// Create or update the underlying autoscaler implementation.
+	// Delete unused autoscaler implementations--there can be only one!
+	switch kpa.Class() {
+	case "kpa":
```

Comment:
tl;dr We are going to want separate `Reconciler` implementations for each "class".

These two implementations share effectively nothing, so I'd much rather see them manifest as separate controllers wrapping distinct Reconcilers.  I would go so far as to say they even belong in distinct packages.

Reply:
I thought along the same lines, but thought it's fine to keep it in one place as long as it's only 2 types. If we keep growing that list, we'd need to think about breaking it up properly.

Reply:
SGTM.

Reply:
Actually, do we gain anything from having resources shared?

Reply:
> having resources shared?

What do you mean?  It's good to have the common PodAutoscaler resource (https://github.com/knative/serving/blob/master/pkg/apis/autoscaling/v1alpha1/kpa_types.go).  That's the "resource" I was talking about.

Reply:
Yeah, there's not really much to share between the implementations.  If there is, I'll put it there.  Mostly I just wanted an intermediate package to put the two implementations under the same folder.

Reply:
I've gotten started on implementing this as two separate reconcilers: https://github.com/knative/serving/pull/2457  (It was just easier to start over).  Take a look to see the direction I'm taking.  Still to do is the webhook validation for immutable `class` and `metrics` annotations and unit tests.